models/gif_cond_exp_multisynapse.cpp

```diff
@@ -0,0 +1,654 @@
+ * Default constructors defining default parameters and state
+ * ---------------------------------------------------------------- */
+
+nest::gif_cond_exp_multisynapse::Parameters_::Parameters_()
+  : g_L_( 4.0 )        // nS
+  , E_L_( -70.0 )      // mV
+  , V_reset_( -55.0 )  // mV
+  , Delta_V_( 0.5 )    // mV
+  , V_T_star_( -35 )   // mV
+  , lambda_0_( 0.001 ) // 1/ms
+  , t_ref_( 4.0 )      // ms
+  , c_m_( 80.0 )       // pF
+  , tau_stc_()         // ms
+  , q_stc_()           // nA
+  , tau_sfa_()         // ms
+  , q_sfa_()           // mV
+  , tau_syn_()         // ms
+  , I_e_( 0.0 )        // pA
+  , receptor_types_()
+  , num_of_receptors_( 0 )
+  , E_ex_( 0.0 )   // mV
+  , E_in_( -85.0 ) // mV
+  , has_connections_( false )
+{
+}
+
+nest::gif_cond_exp_multisynapse::State_::State_( const Parameters_& p )
+  : y_( NUMBER_OF_FIXED_STATES_ELEMENTS, 0.0 )
+  , y0_( 0.0 )
+  , sfa_( 0.0 )
+  , stc_( 0.0 )
+  , sfa_elems_()
+  , stc_elems_()
+  , r_ref_( 0 )
+{
+  y_[ V_M ] = p.E_L_;
+}
+
+nest::gif_cond_exp_multisynapse::State_::State_( const State_& s )
+  : y0_( s.y0_ )
+  , sfa_( s.sfa_ )
+  , stc_( s.stc_ )
+  , r_ref_( s.r_ref_ )
+{
+  sfa_elems_.resize( s.sfa_elems_.size(), 0.0 );
+  for ( size_t i = 0; i < sfa_elems_.size(); ++i )
+    sfa_elems_[ i ] = s.sfa_elems_[ i ];
+
+  stc_elems_.resize( s.stc_elems_.size(), 0.0 );
+  for ( size_t i = 0; i < stc_elems_.size(); ++i )
+    stc_elems_[ i ] = s.stc_elems_[ i ];
+
+  y_ = s.y_;
+}
+
+nest::gif_cond_exp_multisynapse::State_&
+  nest::gif_cond_exp_multisynapse::State_::
+  operator=( const State_& s )
+{
+  assert( this != &s ); // would be bad logical error in program
+
+  y_ = s.y_;
+
+  sfa_elems_.resize( s.sfa_elems_.size(), 0.0 );
+  for ( size_t i = 0; i < sfa_elems_.size(); ++i )
+    sfa_elems_[ i ] = s.sfa_elems_[ i ];
+
+  stc_elems_.resize( s.stc_elems_.size(), 0.0 );
+  for ( size_t i = 0; i < stc_elems_.size(); ++i )
+    stc_elems_[ i ] = s.stc_elems_[ i ];
+
+  y0_ = s.y0_;
+  sfa_ = s.sfa_;
+  r_ref_ = s.r_ref_;
+  stc_ = s.stc_;
+
+  return *this;
+}
+
+/* ----------------------------------------------------------------
+ * Parameter and state extractions and manipulation functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::gif_cond_exp_multisynapse::Parameters_::get( DictionaryDatum& d ) const
+{
+  def< double >( d, names::I_e, I_e_ );
+  def< double >( d, names::E_L, E_L_ );
+  def< double >( d, names::g_L, g_L_ );
+  def< double >( d, names::C_m, c_m_ );
+  def< double >( d, names::V_reset, V_reset_ );
+  def< double >( d, names::Delta_V, Delta_V_ );
+  def< double >( d, names::V_T_star, V_T_star_ );
+  def< double >( d, names::lambda_0, lambda_0_ * 1000.0 ); // convert to 1/s
+  def< double >( d, names::t_ref, t_ref_ );
+  def< double >( d, names::E_ex, E_ex_ );
+  def< double >( d, names::E_in, E_in_ );
+  def< int >( d, "n_synapses", num_of_receptors_ );
```

Comment:
@heplesser I am wondering about this name ``n_synapses``... I know it is what is used in the original [iaf_psc_alpha_multisynapse](https://github.com/nest/nest-simulator/blob/master/models/iaf_psc_alpha_multisynapse.cpp#L108) model, but maybe we should change this to a more explicit ``num_of_synapses`` name.
And I guess this should go into [nest_names.h](https://github.com/nest/nest-simulator/blob/master/nestkernel/nest_names.h)...